.github/workflows/builds.yml

```diff
@@ -13,15 +13,13 @@ jobs:
 
     - name: Install dependencies
       run: |
-          sudo snap refresh lxd --channel=latest
-          sudo adduser "$USER" lxd
-          sudo lxd init --auto
           sudo snap install charmcraft --classic
 
     - name: Build charm
       run: |
           set -eux
-          sudo -g lxd charmcraft pack
+          # XXX: use --destructive-mode on disposable builders like GitHub Action runners only
+          sudo charmcraft pack --destructive-mode
```

Comment:
I know it's a lot faster, but it doesn't feel like a good idea as it means that whatever weirdness may be done in the Github image (like different python version) will affect the build and so invalidate the result of a proper clean build like the one LP will do later.

Reply:
charmcraft seems to take precautions to not be polluted by the host's environment. It uses venv and the python3 interpreter bundled inside it's snap:

```
#!/bin/bash
set -euo pipefail
# Environment
## Part Environment
export CRAFT_ARCH_TRIPLET="x86_64-linux-gnu"
export CRAFT_TARGET_ARCH="amd64"
export CRAFT_PARALLEL_BUILD_COUNT="1"
export CRAFT_PROJECT_DIR="/home/ubuntu/charm-lxd"
export CRAFT_PART_NAME="charm"
export CRAFT_PART_SRC="/home/ubuntu/charm-lxd/build/parts/charm/src"
export CRAFT_PART_SRC_WORK="/home/ubuntu/charm-lxd/build/parts/charm/src"
export CRAFT_PART_BUILD="/home/ubuntu/charm-lxd/build/parts/charm/build"
export CRAFT_PART_BUILD_WORK="/home/ubuntu/charm-lxd/build/parts/charm/build"
export CRAFT_PART_INSTALL="/home/ubuntu/charm-lxd/build/parts/charm/install"
export CRAFT_OVERLAY="/home/ubuntu/charm-lxd/build/overlay/overlay"
export CRAFT_STAGE="/home/ubuntu/charm-lxd/build/stage"
export CRAFT_PRIME="/home/ubuntu/charm-lxd/build/prime"
## Plugin Environment
## User Environment

set -x
env -i LANG=C.UTF-8 LC_ALL=C.UTF-8 PATH=/snap/charmcraft/904/libexec/charmcraft:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/snap/bin SNAP=/snap/charmcraft/904 SNAP_ARCH=amd64 SNAP_NAME=charmcraft SNAP_VERSION=1.6.0 /snap/charmcraft/904/bin/python3 -I /snap/charmcraft/904/lib/charmcraft/charm_builder.py --charmdir /home/ubuntu/charm-lxd/build/parts/charm/build --builddir /home/ubuntu/charm-lxd/build/parts/charm/install --entrypoint /home/ubuntu/charm-lxd/build/parts/charm/build/src/charm.py -b cryptography -p 'setuptools < 58' -r requirements.txt
```

That said, if those artifacts are what you upload to charmhub.io, then I would share the concern too.

Reply:
I just remembered that what is pushed to charmhub.io is actually built on clean Launchpad builders. As such, what is built on GitHub is just to provide a build-ability verdict.

Reply:
Ideally to be compliant with normal company QA policies, we'd need to test the exact artifact published to the snap store.
This isn't currently practical so we're using the closest thing we can to that by building it the exact same way in this job and testing that instead.

This change may lead to what we test differing from what we publish which is problematic.